content/api/commands/origin.md

```diff
@@ -473,49 +505,115 @@ of
+### Dependencies / Sharing Code
 
-Because of the way in which the callback is transmitted and executed, there are
-certain limitations on what code may be run inside it. In particular, the
-following Cypress commands will throw errors if used in the callback:
+It is not possible to use
+[CommonJS `require()`](https://nodejs.org/en/knowledge/getting-started/what-is-require/)
+or
+[dynamic ES module `import()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports)
+within the callback. However, [`Cypress.require()`](/api/cypress-api/require)
+can be utilized to include [npm](https://www.npmjs.com/) packages and other
+files. It is functionally the same as using
+[CommonJS `require()`](https://nodejs.org/en/knowledge/getting-started/what-is-require/)
+in browser-targeted code.
 
-- `cy.origin()`
-- [`cy.intercept()`](/api/commands/intercept)
-- [`cy.session()`](/api/commands/session)
-- [`cy.server()`](/api/commands/server)
-- [`cy.route()`](/api/commands/route)
-- [`Cypress.Cookies.preserveOnce()`](/api/cypress-api/cookies)
+```js
+cy.origin('somesite.com', () => {
+  const _ = Cypress.require('lodash')
+  const utils = Cypress.require('../support/utils')
 
-It is also currently not possible to use
-[`require()`](https://nodejs.org/en/knowledge/getting-started/what-is-require/)
-or
-[dynamic `import()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports)
-within the callback. Because of this limitation, it cannot use
-[npm](https://www.npmjs.com/) packages or other third-party libraries inside the
-callback, as there is no mechanism to reference them. This functionality will be
-provided in a future version of Cypress.
+  // ... use lodash and utils ...
+})
+```
+
+`Cypress.require()` can be used to share custom commands between tests run in
+primary and secondary origins. We recommend this pattern for setting up your
+[support file](/guides/core-concepts/writing-and-organizing-tests#Support-file)
+and setting up custom commands to run within the `cy.origin()` callback:
+
+`cypress/support/commands.js`:
 
-While third-party packages are strictly unavailable, it is possible to reuse
-your **own** code between `cy.origin()` callbacks. The workaround is to create a
-custom Cypress command within the secondary origin in a `before` block:
+```js
+Cypress.Commands.add('clickLink', (label) => {
+  cy.get('a').contains(label).click()
+})
+```
+
+`cypress/support/e2e.js`:
+
+```js
+// makes custom commands available to all Cypress tests, outside of
+// cy.origin() callbacks
+import './commands'
+
+// code we only want run per test, so it shouldn't be run as part of
+// the execution of cy.origin() as well
+beforeEach(() => {
+  // ... code to run before each test ...
+})
+```
+
+`cypress/e2e/spec.cy.js`:
+
+```js
+it('tests somesite.com', () => {
+  cy.origin('somesite.com', () => {
+    // makes custom commands available to all subsequent
```

Comment:
Is this all subsequent calls within this test, within this spec, or within this cypress run?

Reply:
Addressed in https://github.com/cypress-io/cypress-documentation/pull/4747/commits/59809e4535f7b28141413884c0c21767bffeed94